### PR TITLE
2to3

### PR DESCRIPTION
**Pull request by louisabraham**
I used 2to3 to do #93.

### CODE REVIEW

File: cnfformula/cmdline.py [hunk 1]
```diff
@@ -194,7 +194,7 @@ def find_methods_in_package(package,test, sortkey=None):
     result = []
 
     if sortkey == None :
-        sortkey = lambda x : x
+        sortkey = str
```
Comment (louisabraham) [inline cnfformula/cmdline.py]: This is the only manual change I did.

File: cnfformula/__init__.py [hunk 2]
```diff
@@ -29,7 +29,7 @@ def _load_formula_generators():
     # Load the formula generators in the `cnfformula` namespace
     self_ref = sys.modules[__name__]
     self_ref.__dict__.update(loot)
-    __all__.extend(name for name in loot.keys() if name not in __all__)
+    __all__.extend(name for name in list(loot.keys()) if name not in __all__)
```
Comment (marcvinyals) [inline cnfformula/__init__.py]: No need for `list()` here.

Comment (louisabraham) [reply cnfformula/__init__.py]: Indeed, it was done automatically by the module [2to3](https://docs.python.org/3/library/2to3.html). Note that in python 2, keys returns a list whereas iterkeys returns an iterator.

**Comment by marcvinyals** [comment]
Thanks for your PR! There are a few places where an automatic translation is not the best idea. For instance, it adds way too many calls to `list()` (I commented on such an example). In particular, the code often uses `range` instead of `xrange` because at some point we aimed to write code that would work both for Python 2 and Python 3. These calls should be left alone. Is it possible to instruct 2to3 to do so?

**Comment by louisabraham** [comment]
From the [doc](https://docs.python.org/3/library/2to3.html#fixers):

> Each step of transforming code is encapsulated in a fixer. The command 2to3 -l lists them. As documented above, each can be turned on and off individually.

However, since `range` returns lists in Python 2, I don't see why it bothers to call `list` in Python 3, as the performance will be similar to the Python 2 version. I don't think one can beat manual checking.

**Comment by MassimoLauria** [comment]
I will close this issue since I merged #96 

Thank oyu both for your help and effort. This upgrade was long due.

**Pull request closed without merging**